### ISSUE TITLE
Next Fest October 2022

### ISSUE THREAD

**Issue by evgewakan** [chr233/ASFEnhance#31]
https://store.steampowered.com/sale/nextfest

can you add command/plugin for it? need view special explore list. 6 time = 6 lvl = 100 exp + badge for profile.
![1234](https://user-images.githubusercontent.com/74374112/193754640-4cca150c-3675-4c8a-82d9-a86b4f364d9a.PNG)

**Comment by chr233** [comment]
I'm working on it, it needs some time

**Comment by evgewakan** [comment]
how we can support you? link your trade offer

**Comment by woctezuma** [comment]
> how we can support you? link your trade offer

There is a button for this purpose in the README.

**Comment by adminerest** [comment]
I wrote a python script for you guys as a reference
```

# filename: main.py

import event_pb2
from requests import Session
from re import findall
from base64 import encodebytes

session = Session()

STEAM_LOGIN_SECURE = ''.replace('%7C', '|')

session.cookies.set('steamLoginSecure', STEAM_LOGIN_SECURE)

rp = session.get('https://store.steampowered.com/sale/nextfest')
access_token = findall(r'data-loyalty_webapi_token="&quot;[0123456789abcdef]+&quot;"', rp.text)[0].split('&quot;')[1]

for _ in range(6):
    discovery_queue_rp = event_pb2.GetDiscoveryQueueResponse()
    params = {
        'access_token': access_token,
        'input_protobuf_encoded': 'CAESAkdCGAEwAWIGCgQI/7VL'
    }
    rp = session.get('https://api.steampowered.com/IStoreService/GetDiscoveryQueue/v1', params=params)
    discovery_queue_rp.ParseFromString(rp.content)
    for appid in discovery_queue_rp.appid:
        skip_discovery_queue_request = event_pb2.SkipDiscoveryQueueItemRequest()
        skip_discovery_queue_request.queue_type = 1
        skip_discovery_queue_request.appid = appid
        skip_discovery_queue_request.store_page_filter.unknown_a.unknown_b = 1235711
    
        params = {
            'access_token': access_token
        }
        data = {
            'input_protobuf_encoded': encodebytes(skip_discovery_queue_request.SerializeToString()).decode().strip()
        }
        rp = session.post('https://api.steampowered.com/IStoreService/SkipDiscoveryQueueItem/v1', params=params, data=data)
        print(rp.status_code)
```

```
# -*- coding: utf-8 -*-
# Generated by the protocol buffer compiler.  DO NOT EDIT!
# source: event.proto
# filename: event_pb2.py

"""Generated protocol buffer code."""
from google.protobuf.internal import builder as _builder
from google.protobuf import descriptor as _descriptor
from google.protobuf import descriptor_pool as _descriptor_pool
from google.protobuf import symbol_database as _symbol_database
# @@protoc_insertion_point(imports)

_sym_db = _symbol_database.Default()




DESCRIPTOR = _descriptor_pool.Default().AddSerializedFile(b'\n\x0b\x65vent.proto\"o\n\x1dSkipDiscoveryQueueItemRequest\x12\x12\n\nqueue_type\x18\x01 \x02(\x05\x12\r\n\x05\x61ppid\x18\x02 \x02(\x05\x12+\n\x11store_page_filter\x18\x03 \x02(\x0b\x32\x10.StorePageFilter\"6\n\x0fStorePageFilter\x12#\n\tunknown_a\x18\x01 \x02(\x0b\x32\x10.UnknownMessage1\"$\n\x0fUnknownMessage1\x12\x11\n\tunknown_b\x18\x01 \x02(\x05\"\x9b\x01\n\x19GetDiscoveryQueueResponse\x12\r\n\x05\x61ppid\x18\x01 \x03(\x05\x12\x14\n\x0c\x63ountry_code\x18\x02 \x02(\t\x12\x17\n\x0funknown_string1\x18\x03 \x02(\t\x12\x14\n\x0cunknown_int1\x18\x04 \x02(\x05\x12\x14\n\x0cunknown_int2\x18\x05 \x02(\x05\x12\x14\n\x0cunknown_int3\x18\x06 \x02(\x05')

_builder.BuildMessageAndEnumDescriptors(DESCRIPTOR, globals())
_builder.BuildTopDescriptorsAndMessages(DESCRIPTOR, 'event_pb2', globals())
if _descriptor._USE_C_DESCRIPTORS == False:

  DESCRIPTOR._options = None
  _SKIPDISCOVERYQUEUEITEMREQUEST._serialized_start=15
  _SKIPDISCOVERYQUEUEITEMREQUEST._serialized_end=126
  _STOREPAGEFILTER._serialized_start=128
  _STOREPAGEFILTER._serialized_end=182
  _UNKNOWNMESSAGE1._serialized_start=184
  _UNKNOWNMESSAGE1._serialized_end=220
  _GETDISCOVERYQUEUERESPONSE._serialized_start=223
  _GETDISCOVERYQUEUERESPONSE._serialized_end=378
# @@protoc_insertion_point(module_scope)
```

```

// proto needed
syntax = "proto2";

message SkipDiscoveryQueueItemRequest {
  required int32 queue_type = 1;
  required int32 appid = 2;
  required StorePageFilter store_page_filter = 3;
}
message StorePageFilter {
  required UnknownMessage1 unknown_a = 1;
}
message UnknownMessage1 {
  required int32 unknown_b = 1;
}

message GetDiscoveryQueueResponse {
  repeated int32 appid = 1;
  required string country_code = 2;
  required string unknown_string1 = 3;
  required int32 unknown_int1 = 4;
  required int32 unknown_int2 = 5;
  required int32 unknown_int3 = 6;
}
```

**Comment by chr233** [comment]
> how we can support you? link your trade offer

I put my tread link in README.md

**Comment by evgewakan** [comment]
> I put my tread link in README.md

yes, I already found it, thanks for the work, I sent a small present

**Comment by chr233** [comment]
> > I put my tread link in README.md
> 
> yes, I already found it, thanks for the work, I sent a small present

tkx a lot

**Comment by chr233** [comment]
I have trouble with my dotnet dev environment, trying re-install vs2022, steam festival features are writen, but not work properly, just need more time to fix (after install the vs2022 complete)

**Comment by chr233** [comment]
All work done, use `AU` to update automaticly or download manually via https://github.com/chr233/ASFEnhance/releases/tag/1.6.12.705

**Comment by chr233** [comment]
> ~All work done, use `AU` to update automaticly or download manually via https://github.com/chr233/ASFEnhance/releases/tag/1.6.12.705~
> 
> maybe somewhere still not work, checking

![image](https://user-images.githubusercontent.com/13798576/193983930-2816cefc-b744-4f32-b963-5d93403aee3c.png)

**Comment by chr233** [comment]
Now fixed, use `1.6.12.717` instead.
